xds/src/main/java/io/grpc/xds/WeightedRoundRobinLoadBalancer.java

```diff
@@ -267,7 +273,7 @@ private void updateWeight() {
         }
       }
       if (weightedChannelCount < 2) {
-        rrMode = true;
+        scheduler = null;
```

Comment:
Let's not do this either. Always use the scheduler. The performance benefit doesn't matter (it is optimizing a case we don't care about to be "faster" than the case we do care about) and it just opens up the possibility for bugs.